### ISSUE TITLE
Document rbenv exec

### ISSUE THREAD

**Issue by tremby** [rbenv/rbenv#1182]
I'm a Rubynewbie.

In trying to get an existing project set up on my local system, I installed rbenv according to the instructions (I think -- I had a few bits of trouble but ended up in a state where I think it's working correctly), installed the version I need, set it as local, installed bundler, then got stuck trying to *run* `bundle`.

It was running the systemwide version, not the local version. I considered adding the local ruby version's gem bin path to my PATH variable but that felt wrong -- isn't rbenv supposed to let me use different versions in different places and not worry about that sort of thing?

Eventually I wondered if there was equivalent of node's `npx`, and looked at `rbenv help` and didn't see anything which looked right.

It wasn't until I used the tab-completion rbenv installed on my shell that I saw the `exec` command. Then I tried `rbenv exec` and lo and behold, that looks exactly like what I want. So `rbenv exec bundle install` did what I wanted.

That took me far too long to find. It's not in `rbenv help`, and it's not in the https://github.com/rbenv/rbenv#command-reference or in the readme at all.

Is this an oversight? Or is it possible I set something up wrong, and I shouldn't need to do this at all?

**Comment by mislav** [comment]
`rbenv-exec` is an internal command and, as such, we're trying to discourage interfacing with it directly. That's why it's not clearly documented.

Our philosophy is that you should never need to run `rbenv exec bundle` yourself. You should only ever run `bundle`, and that should activate the Bundler version that belongs to the Ruby version selected for the current project. The rbenv project doesn't have nor need an `npx` equivalent.

If that didn't work for you, it's possible that `~/.rbenv/shims/bundle` didn't exist for you, or that the `~/.rbenv/shims` isn't prepended to your PATH (see https://github.com/rbenv/rbenv-installer#rbenv-doctor).

In short:
1. Ensure that the current project has a local version selected, e.g. `rbenv local 2.5.5`;
2. Ensure that Bundler is installed for that version, e.g. `gem install bundler`;
3. Running `bundle` should now execute Bundler in Ruby 2.5.5.

**Comment by tremby** [comment]
OK. I'm not sure what I was doing wrong, but I was certain at least at the time that I had followed all steps correctly, and I know how `PATH` and `which` etc work. (Or at least I thought I did.) Anyway, it's working just fine now, finding `bundle` via the shims. Sorry for the noise.
